### PR TITLE
[PARQUET-1500] Replace Closeables with try-with-resources

### PR DESCRIPTION
**Pull request by Fokko**
https://jira.apache.org/jira/browse/PARQUET-1500

### CODE REVIEW

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java [hunk 1]
```diff
@@ -53,22 +53,14 @@ private SerializationUtil() { }
...
-    try {
-      baos = new ByteArrayOutputStream();
-      gos = new GZIPOutputStream(baos);
-      oos = new ObjectOutputStream(gos);
-      oos.writeObject(obj);
-    } finally {
-      Closeables.close(oos);
-      Closeables.close(gos);
-      Closeables.close(baos);
+    try(ByteArrayOutputStream baos = new ByteArrayOutputStream()) {
```
Comment (ryanrupp) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java]: You can consolidate these into a single `try` e.g.:

```
try (ByteArrayOutputStream baos = new ByteArrayOutputStream();
       GZIPOutputStream gos = new GZIPOutputStream(baos);
       ObjectOutputStream oos = new ObjectOutputStream(gos)) {
    oos.writeObject(obj);
    conf.set(key, new String(Base64.encodeBase64(baos.toByteArray()), "UTF-8"));
}
```

Comment (Fokko) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java]: Sweet suggestion, thanks!

**Comment by nandorKollar** [comment]
@Fokko looks like semantic versioning plugin didn't like the removal of Closeables. How about deprecating it instead of removing so that this addition could get committed to 1.x line?

**Comment by Fokko** [comment]
Thanks for the clarification @nandorKollar. The CI error was a bit cryptic to me, but now it makes sense. I've added a deprecation warning instead of removing the class.

File: parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java [hunk 2]
```diff
@@ -53,22 +53,13 @@ private SerializationUtil() { }
...
+      try(GZIPOutputStream gos = new GZIPOutputStream(baos);
+            ObjectOutputStream oos = new ObjectOutputStream(gos)) {
+        oos.writeObject(obj);
+      }
+      conf.set(key, new String(Base64.encodeBase64(baos.toByteArray()), StandardCharsets.UTF_8));
```
Comment (gszadovszky) [inline parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java]: This code is not equivalent to the original one: `baos.toByteArray()` was invoked after closing the stream, now it is the other way around.

Comment (Fokko) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java]: This is true, but with the try-with-resource, we don't have a reference after closing the stream. Please note that the `.close()` on the `ByteArrayOutputStream` doesn't have any effect: https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#close()
The important thing here is that the `ObjectOutputStream` and `GZIPOutputStream` are flushed (and closed).

Comment (gszadovszky) [reply parquet-hadoop/src/main/java/org/apache/parquet/hadoop/util/SerializationUtil.java]: You're right, thanks for explaining.

**Pull request merged**